proselint/command_line.py

```diff
@@ -1,94 +1,163 @@
-    dirname = os.path.dirname
-    corpus_path = os.path.join(
-        dirname(dirname(os.path.realpath(__file__))), "corpora", corpus)
-    start = time.time()
-    for file in os.listdir(corpus_path):
-        filepath = os.path.join(corpus_path, file)
-        if filepath[-3:] == ".md":
-            subprocess.call(["proselint", filepath, ">/dev/null"])
-
-    return time.time() - start
-
-
-@click.command(context_settings=CONTEXT_SETTINGS)
-@click.version_option(__version__, '--version', '-v', message='%(version)s')
-@click.option('--config', is_flag=False,
-              type=click.Path(exists=True, path_type=Path),
-              help="Path to configuration file.")
-@click.option('--debug', '-d', is_flag=True, help="Give verbose output.")
-@click.option('--json', '-j', 'output_json', is_flag=True,
-              help="Output as JSON.")
-@click.option('--time', '-t', is_flag=True, help="Time on a corpus.")
-@click.option('--demo', is_flag=True, help="Run over demo file.")
-@click.option('--compact', is_flag=True, help="Shorten output.")
-@click.option('--dump-config', is_flag=True, help="Prints current config.")
-@click.option('--dump-default-config', is_flag=True,
-              help="Prints default config.")
-@click.argument('paths', nargs=-1, type=click.Path(exists=True, path_type=Path))
-def proselint(
-    paths: list[Path], config=None, version=None,
-    debug=None, output_json=None, time=None, demo=None, compact=None,
-    dump_config=None, dump_default_config=None
-) -> None:
+if TYPE_CHECKING:
+    from types import FrameType
+
+
+class ExitStatus(IntEnum):
+    """Exit status for proselint's command line."""
+
+    SUCCESS = 0
+    SUCCESS_ERR = 1
+    UNACCEPTED_ARGS = 2
+    INTERRUPT = 3
+
+
+def interrupt_handler(signalnum: int, _frame: Optional[FrameType]) -> None:
+    """Exit proselint gracefully from an interrupt."""
+    log.warning(f"\nExiting (received {Signals(signalnum).name} {signalnum}).")
+    sys.exit(ExitStatus.INTERRUPT)
+
+
+def get_parser() -> ArgumentParser:
+    """Create an `ArgumentParser` for command line arguments."""
+    global_parser = ArgumentParser()
+    global_parser.add_argument(
+        "--config",
+        type=Path,
+        help="path to a configuration file (`.proselintrc.json`)",
+    )
+    global_parser.add_argument(
+        "--output-format",
+        "-o",
+        choices=[output_format.value for output_format in OutputFormat],
+        default=OutputFormat.FULL,
+        type=OutputFormat,
+        help="the format to display results in",
+    )
+    global_parser.add_argument(
+        "--verbose", "-v", action="store_true", help="enable verbose logging"
+    )
+
+    parser = ArgumentParser(
+        parents=(global_parser,),
+        conflict_handler="resolve",
+        description="proselint, a linter for prose.",
+        usage="%(prog)s [options] <command>",
+    )
+    subparsers = parser.add_subparsers(
+        title="commands", dest="subcommand", metavar=""
```

Comment:
```suggestion
        title="commands", dest="subcommand", metavar="\033[K\033[F"
```
This could be used to remove the new blank line where ANSI escape sequences are supported, but it's a hacky solution. It would be better to leave it as-is, short of making a custom help formatter.